optimizer hints: add docs for hint SET_VAR (#14790)
pingcap/docs: system-variables.md

>
> Variables using the same unit might compete for the same set of resources.

## Hint SET_VAR

> TiDB introduces the Opitmizer Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) to control the execution behavior of one SQL more flexiable. This hint can change the value of specified variables at SQL level.

Reviewer: ```suggestion
TiDB introduces the Opitmizer Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) that changes the value of the specified system variable at statement runtime without changing the value in current session.  "Applies to Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)): Yes" indicates the particular variable can be set by [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)). 
```

>
> Variables using the same unit might compete for the same set of resources.

Starting from v7.4.0, you can temporarily modify the value of some `SESSION` variables during statement execution using [`SET_VAR`](/optimizer-hints.md#set_varvar_namevar_value). After the statement is executed, the value of the system variable in the current session is automatically changed back to the original value. This hint can be used to modify some system variables related to the optimizer and executor. Variables in this document have a `Applies to hint SET_VAR` setting, which can be configured to `Yes` or `No`.

- For variables with the `Applies to hint SET_VAR: Yes` setting, you can use the [`SET_VAR`](/optimizer-hints.md#set_varvar_namevar_value) hint to modify the value of the system variable in the current session during statement execution.
- For variables with the `Applies to hint SET_VAR: No` setting, you cannot use the [`SET_VAR`](/optimizer-hints.md#set_varvar_namevar_value) hint to modify the value of the system variable in the current session during statement execution.

For more information about the `SET_VAR` hint, see [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value).

> ## Hint SET_VAR
> 
> TiDB introduces the Opitmizer Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) to control the execution behavior of one SQL more flexiable. This hint can change the value of specified variables at SQL level.
> The following chapter will list which variables can be controlled by the hint.
> 

Reviewer 2: ```suggestion
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Boolean
- Default value: `ON`
- This variable controls whether to use the high precision mode when computing the window functions.

> 

Reviewer 2: ```suggestion
```
